libraries/Microsoft.Bot.Connector/Authentication/MicrosoftAppCredentials.cs

```diff
@@ -210,6 +292,27 @@ public OAuthException(string body, Exception inner)
             }
         }
 
+        /// <summary>
+        /// Has the token expired?  If so, then we await on every attempt to get a new token
+        /// </summary>
+        /// <param name="token"></param>
+        /// <returns></returns>
+        private bool IsTokenExpired(OAuthResponse token)
+        {
+            return DateTime.UtcNow > token.expiration_time;
+        }
+
+        /// <summary>
+        /// has token reached half/life ? If so, we get more agressive about trying to refresh it in the background
+        /// </summary>
+        /// <param name="token"></param>
+        /// <returns></returns>
+        private bool IsTokenOld(OAuthResponse token)
```

Comment:
Same comments for this as those on `IsTokenExpired`. If you don't feel comfortable inlining the math then you can just skip the expression bodied method part.